camel-frappe-api/src/main/java/com/ozonehis/camel/frappe/sdk/internal/security/DefaultFrappeAuthentication.java

```diff
@@ -33,10 +41,31 @@ public Response intercept(@NotNull Chain chain) throws IOException {
     }
 
     public List<String> getSessionCookie(Request incomingRequest) throws IOException {
+        WrappedCookie sessionCookie = cookieCache.get(SESSION_COOKIE_NAME);
+        if (sessionCookie != null && !sessionCookie.isExpired()) {
+            log.debug("Session cookie found and not expired. Using it...");
+            return sessionCookie.unwrap();
+        } else {
+            log.debug("Session cookie not found or expired. Logging in...");
+            return login(incomingRequest, cookieCache);
+        }
+    }
+
+    private List<String> login(Request incomingRequest, CookieCache cookieCache) throws IOException {
         var baseUrl = getBaseUrl(incomingRequest);
         Request request = buildLoginRequest(baseUrl);
         try (Response response = executeRequest(request)) {
-            return extractCookies(response);
+            var cookies = response.headers("set-cookie");
+            var cookieList = Cookie.parseAll(incomingRequest.url(), response.headers());
+            cookieList.forEach(cookie -> {
+                if (cookie.name().equalsIgnoreCase("sid") && !cookie.value().isEmpty()) {
+                    cookieCache.clearExpired();
+                    cookieCache.put(SESSION_COOKIE_NAME, new WrappedCookie(cookie.expiresAt(), cookies));
+                }
+            });
```

Comment:
> As I see it, the scheme here sort of works if the server can ignore invalid things

Yes, it's work because server ignores the invalid things. Let update as per spec